app/controllers/users/two_factor_setup/authenticator_setup_controller.rb

```diff
@@ -0,0 +1,81 @@
+module Users
+  module TwoFactorSetup
+    class AuthenticatorSetupController < TwoFactorVerificationController
+      before_action :authenticate_user!
```

Comment:
I _think_ you can put `before_action :authenticate_user!` into TwoFactorVerificationController, and remove it from the other two_factor_setup controllers?